### ISSUE TITLE
Consistency in naming functions in proj.h for memory deallocation

### ISSUE THREAD

**Issue by snowman2** [OSGeo/PROJ#1198]
It would be nice to use a single term for deallocation in the functions in proj.h. Currently there is:
- destroy
- unref
- free
And probably others. It would be nice to choose one for consistency.

**Comment by kbevers** [comment]
Yes, I agree and this is something that we should have a look at before release time comes. There's still over three months left to sort this stuff out and at the moment other things have higer priority.

Note that some of the odd naming choices come from the fact that we have quite a lot of restrictions from the older API's that would cause clashing names. That's the reason for the `_destroy` functions - `free` was already occupied at the time, so to avoid confusing we opted to use a new word for the deallocation function(s) in `proj.h`.

**Comment by rouault** [comment]
As those functions are heavily used it would be best to decide now if they must be changed.
The _unref() things were my introduction. The rationale was that the underlying C++ object might not be immediately destroyed, if it is actually owned by another alive object. But I realize now that the C wrapping structure is really invalidated after _unref(), and should no longer be used, so from an external point of view, this is a real destruction.
I've also added a few _free() for a few simple structures like a list of int and list of strings.
So basically there was only _destroy() before in proj.h. I'm fine to change my new _unref() and _free() to _destroy() as well.

**Comment by kbevers** [comment]
>  I'm fine to change my new _unref() and _free() to _destroy() as well.

Okay, I think you should do that then. Thanks!

**Comment by snowman2** [comment]
🎉
